### PR TITLE
Merging new branch annotations_dump (which has xml to json conversion script) to master

### PR DESCRIPTION
**Pull request by KeshavBalaji**
Adding standalone Python script that parses xml file of annotations and writes to corresponding json file

### CODE REVIEW

File: xml_to_json.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
+"""
+File: xml_to_json.py
+Author: Keshav Balaji
+Description: Standalone Python script to parse an xml file of annotations and write
+the data to a corresponding json file
+"""
+
+import os
+import glob
```
Comment (iross) [inline xml_to_json.py]: Unused  -- can be removed

File: xml_to_json.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
+"""
+File: xml_to_json.py
+Author: Keshav Balaji
+Description: Standalone Python script to parse an xml file of annotations and write
+the data to a corresponding json file
+"""
+
+import os
```
Comment (iross) [inline xml_to_json.py]: Unused -- can be removed

File: xml_to_json.py [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+import glob
+import json
+import xmltodict
+import sys
+from pydantic import BaseModel, Field
+from typing import List
+from numpy import arange
```
Comment (iross) [inline xml_to_json.py]: Unused -- can be removed

File: xml_to_json.py [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+    def union(self, other: 'AnnotationBounds'):
+        """ Get the union area of two rectangles """
+        return self.area() + other.area() - self.intersection(other)
+
+
+    def intersection_over_union(self, other: 'AnnotationBounds'):
+        """ Get the intersection-over-union of two rectangles """
+        return self.intersection(other) / self.union(other)
+
+def parse_xml() -> list[AnnotationBounds]:
```
Comment (iross) [inline xml_to_json.py:51]: Add a docstring to explain what inputs and outputs this method expects and returns

File: xml_to_json.py [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+    # for xml_path in glob.glob(annotations_path):
+    with open(sys.argv[1]) as xml_data:
+        xml_dict = xmltodict.parse(xml_data.read())
+        page_num = int(xml_dict['annotation']['page'])
+        objects = xml_dict['annotation']['object']
+        annotations.extend(_cosmos_obj_from_manual_obj(page_num, obj) for obj in objects)
+
+    return annotations, xml_data.name[:-4]
+
+def _cosmos_obj_from_manual_obj(page_num, obj):
```
Comment (iross) [inline xml_to_json.py]: Typically an underscore in a method name indicates that it is intended for private use (e.g. within a class) -- see https://realpython.com/python-double-underscore/

File: xml_to_json.py [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+        bounds = [int(float(b)) 
+            for b in [bbox['xmin'],bbox['ymin'],bbox['xmax'],bbox['ymax']]]
+        
+        return AnnotationBounds(
+            page_num = page_num,
+            postprocess_cls = obj['name'],
+            bounding_box= bounds
+        )
+
+def create_json(xml_annotations, filename):
```
Comment (iross) [inline xml_to_json.py:81]: Add a docstring here too

File: xml_to_json.py [hunk 7]
```diff
@@ -0,0 +1,86 @@
+"""
+File: xml_to_json.py
+Author: Keshav Balaji
+Description: Standalone Python script to parse an xml file of annotations and write
+the data to a corresponding json file
```
Comment (iross) [inline xml_to_json.py:6]: A usage example is useful here to show what it expects as inputs defined at the command line

File: xml_to_json.py [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+import glob
+import json
+import xmltodict
+import sys
+from pydantic import BaseModel, Field
+from typing import List
+from numpy import arange
+
+DEFAULT_REGION_TYPES = ["Figure", "Equation", "Table"]
+# BASE_DIR = os.path.abspath(os.path.join(os.path.dirname(__file__),'../..'))
```
Comment (iross) [inline xml_to_json.py]: Let's go ahead and delete this kind of code remnant.

**Review by iross — COMMENTED** [review]
Looks good! I left a few comments. I'll leave this open for now -- let's also put the  XML reorganization script on this branch and repo as well.

**Comment by KeshavBalaji** [comment]
Merging annotations_dump branch with updated xml_to_json.py and organize_ground_truths.py to master

**Pull request merged**